integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableColRenameDataTypeChangeCommand.scala

```diff
@@ -0,0 +1,324 @@
+        carbonTable,
+        tableInfo,
+        addColumnSchema,
+        schemaEvolutionEntry)
+      val alterTableColRenameAndDataTypeChangePostEvent
+      : AlterTableColRenameAndDataTypeChangePostEvent =
+        AlterTableColRenameAndDataTypeChangePostEvent(sparkSession, carbonTable,
+          alterTableColRenameAndDataTypeChangeModel)
+      OperationListenerBus.getInstance
+        .fireEvent(alterTableColRenameAndDataTypeChangePostEvent, operationContext)
+      if (isDataTypeChange) {
+        LOGGER
+          .info(s"Alter table for column rename or data type change is successful for table " +
+                s"$dbName.$tableName")
+      }
+      if (alterTableColRenameAndDataTypeChangeModel.isColumnRename) {
+        LOGGER.info(s"Alter table for column rename is successful for table $dbName.$tableName")
+      }
+    } catch {
+      case e: Exception =>
+        if (carbonTable != null) {
+          AlterTableUtil
+            .revertColumnRenameAndDataTypeChanges(dbName, tableName, timeStamp)(sparkSession)
+        }
+        if (isDataTypeChange) {
+          throwMetadataException(dbName, tableName,
+            s"Alter table data type change operation failed: ${ e.getMessage }")
+        } else {
+          throwMetadataException(dbName, tableName,
+            s"Alter table data type change or column rename operation failed: ${ e.getMessage }")
+        }
+    } finally {
+      // release lock after command execution completion
+      AlterTableUtil.releaseLocks(locks)
+    }
+    Seq.empty
+  }
+
+  /**
+   * This method update the schema info and refresh the table
+   *
+   * @param sparkSession
+   * @param carbonTable              carbonTable
+   * @param tableInfo                tableInfo
+   * @param addColumnSchema          added column schema
+   * @param schemaEvolutionEntryList new SchemaEvolutionEntry
+   */
+  private def updateSchemaAndRefreshTable(sparkSession: SparkSession,
+      carbonTable: CarbonTable,
+      tableInfo: TableInfo,
+      addColumnSchema: ColumnSchema,
+      schemaEvolutionEntry: SchemaEvolutionEntry): Unit = {
+    val schemaConverter = new ThriftWrapperSchemaConverterImpl
+    val a = List(schemaConverter.fromExternalToWrapperColumnSchema(addColumnSchema))
+    val (tableIdentifier, schemaParts, cols) = AlterTableUtil.updateSchemaInfo(
+      carbonTable, schemaEvolutionEntry, tableInfo, Some(a))(sparkSession)
+    sparkSession.sessionState.catalog.asInstanceOf[CarbonSessionCatalog]
+      .alterColumnChangeDataType(tableIdentifier, schemaParts, cols)
+    sparkSession.catalog.refreshTable(tableIdentifier.quotedString)
+  }
+
+  /**
+   * This method will validate a column for its data type and check whether the column data type
+   * can be modified and update if conditions are met.
+   */
+  private def validateColumnDataType(
+      dataTypeInfo: DataTypeInfo,
+      carbonColumn: CarbonColumn): Unit = {
+    carbonColumn.getDataType.getName match {
```

Comment:
it is better to if check on `carbonColumn.getDataType` instead of matching strings

Reply:
this logic is kept same as old , as it was for datatype change command, `carbonColumn.getDataType` gives object and with that also we need to compare the string from datatypeinfo object as it has just a string, so this is ok i think